cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/CreateMoneyFlow.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.flow
+
+import co.paralleluniverse.fibers.Suspendable
+import com.example.contract.CashContract
+import com.example.state.CashState
+import com.example.utils.bankProvider
+import net.corda.core.contracts.Command
+import net.corda.core.contracts.requireThat
+import net.corda.core.flows.*
+import net.corda.core.identity.Party
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.TransactionBuilder
+
+object CreateMoneyFlow {
+    @InitiatingFlow
+    @StartableByRPC
+    class Initiator(val cashValue: Long, val moneyOwner: Party) : FlowLogic<SignedTransaction>() {
+        @Suspendable
+        override fun call(): SignedTransaction {
+            val notary = serviceHub.networkMapCache.notaryIdentities[0]
+            val bank = serviceHub.bankProvider()
+
+            require(ourIdentity.owningKey == bank.owningKey) { "only bank can create money" }
```

Comment:
I know what you are trying to achieve here, but I don't think it is useful. You see, if I am malicious, and want to issue Xavier dollars, then I can just create my own `CreateMoneyFlow`, and boom I have created dollars. What matters is when people **receive** the dollars, they will sign the transaction only if it is what they expect.